src/rpc/actor_registry.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright 2019-2025 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+use crate::lotus_json::HasLotusJson;
+use crate::networks::ACTOR_BUNDLES_METADATA;
+use crate::shim::actors::{
+    AccountActorStateLoad, CronActorStateLoad, EVMActorStateLoad, MarketActorStateLoad,
+    MinerActorStateLoad, account, cron, evm, market, miner,
+};
+use crate::shim::machine::BuiltinActor;
+use ahash::{HashMap, HashMapExt};
+use anyhow::anyhow;
+use cid::Cid;
+use fvm_ipld_blockstore::Blockstore;
+use once_cell::sync::Lazy;
+use serde_json::Value;
+
+// Build a map from CIDs to actor types and versions once at startup
+static CID_TO_ACTOR_TYPE: Lazy<HashMap<Cid, (BuiltinActor, u64)>> = Lazy::new(|| {
+    let mut map = HashMap::new();
+
+    // Populate the map using data from ACTOR_BUNDLES_METADATA
+    for ((_, _), metadata) in ACTOR_BUNDLES_METADATA.iter() {
```

Comment:
I'm not sure if this works for miner states. You might need to iterate over all actor states just like what state migrator does.

Reply:
Or maybe just try deserializing into each actor state type, like how serde works for untagged enum

Reply:
Further comments on this: https://filecoinproject.slack.com/archives/C029LPZ5N73/p1747393581932789